lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -739,6 +742,9 @@ def __init__(self, collection_name, subdirs, resource, ref_type):
         self.n_python_collection_package_name = to_native('.'.join(package_components))
 
         if self.ref_type == u'role':
+            if not self.is_valid_role_name(resource):
+                display.warning("Role names are now limited to contain only lowercase alphanumeric characters, "
+                                "plus _ and start with an alpha character. Please check %s" % resource)
```

Comment:
i would rephrase a bit, 'roles names in collections are limited' .. standalone roles are not and collections always had this limitation.

Reply:
probably should just be an error as we won't be able to find the role and later on we'll fail with 'role not found'